R/validate.R

```diff
@@ -39,13 +39,22 @@ validate_dm <- function(x) {
 
   dm_col_names <- set_names(map(def$data, colnames), table_names)
 
-  fks <- c_list_of(def$fks)
+  fks <-
+    def %>%
+    select(ref_table = table, fks) %>%
+    unnest_list_of_df("fks")
+
   check_fk_child_tables(fks$table, table_names)
 
   fks %>%
     unnest_col("column", character()) %>%
     check_colnames(dm_col_names, "FK")
 
+  fks %>%
+    unnest_col("ref_column", character()) %>%
+    select(table = ref_table, column = ref_column) %>%
+    check_colnames(dm_col_names, "Parent key")
+
```

Comment:
Do we need to keep the original code too, because the fks have table and ref-table? (How do we test `dm_validate()` ?)

Reply:
I thought both would work this way but better test indeed. I think the easiest is to take a valid dm, edit its data artificially then test with expect_error, I ll update the PR

Reply:
2 tests were added to test for respective existence of `column` and `ref_column`